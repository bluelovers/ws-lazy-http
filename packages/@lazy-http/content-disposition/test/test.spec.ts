import contentDisposition from '..';
import { parse } from '..';

describe(`content-disposition`, () =>
{

	test(`contentDisposition`, () =>
	{
		let actual = contentDisposition(`小說名字.epub`);
		let expected = `attachment; filename="%E5%B0%8F%E8%AA%AA%E5%90%8D%E5%AD%97.epub"; filename*=UTF-8''%E5%B0%8F%E8%AA%AA%E5%90%8D%E5%AD%97.epub`;

		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();

	});

	test(`parse`, () =>
	{
		let actual = parse(`attachment; filename="%E5%B0%8F%E8%AA%AA%E5%90%8D%E5%AD%97.epub"; filename*=UTF-8''%E5%B0%8F%E8%AA%AA%E5%90%8D%E5%AD%97.epub`);
		let expected = `小說名字.epub`;

		expect(actual).toHaveProperty('value', expected);
		expect(actual).toMatchSnapshot();

	});

})
