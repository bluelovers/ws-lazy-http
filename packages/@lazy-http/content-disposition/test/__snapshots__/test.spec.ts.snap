// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content-disposition contentDisposition 1`] = `"attachment; filename=\\"%E5%B0%8F%E8%AA%AA%E5%90%8D%E5%AD%97.epub\\"; filename*=UTF-8''%E5%B0%8F%E8%AA%AA%E5%90%8D%E5%AD%97.epub"`;

exports[`content-disposition parse 1`] = `
Object {
  "result": Object {
    "filename": "%E5%B0%8F%E8%AA%AA%E5%90%8D%E5%AD%97.epub",
    "filename*": "%E5%B0%8F%E8%AA%AA%E5%90%8D%E5%AD%97.epub",
  },
  "type": "attachment",
  "value": "小說名字.epub",
}
`;
